packages/csharp/PortingAssistant/PortingAssistant.Common/Utils/CustomerContributionUtils.cs

```diff
@@ -0,0 +1,138 @@
+                string metadataPath = path + "/" + "metadata";
+                var metaUploadResponse = ContentsUpload(client, endPoint, metadataPath, new StringContent(contents)).Result;
+                var logUploadResponse = LogUpload(client, path, endPoint);
+                Response<bool, string> response;
+                if (metaUploadResponse.status && logUploadResponse.status)
+                {
+                   response = new Response<bool, string>
+                   {
+                       Status = Response<bool, string>.Success(),
+                       Value = true
+                   };
+                }
+                else
+                {
+                    string msgResponse = string.Format("metadata:{0}\nmain.log:{1}", metaUploadResponse.msg, logUploadResponse.msg);
+                    response = new Response<bool, string>
+                    {
+                        Status = Response<bool, string>.Failed(new Exception(msgResponse)),
+                        ErrorValue = msgResponse
+                    };
+                }
+                return response;
+            }
+            catch (Exception ex)
+            {
+                return new Response<bool, string>
+                {
+                    Status = Response<bool, string>.Failed(ex),
+                    ErrorValue = ex.Message
+                };
+            }
+        }
+
+        public static Response<bool, string> RuleContributionUpload(string path, string contents, string endpoint)
+        {
+            try
+            {
+                var client = new HttpClient();
+                var metaUploadResponse = ContentsUpload(client, endpoint, path, new StringContent(contents)).Result;
+                Response<bool, string> response;
+                if (metaUploadResponse.status)
+                {
+                    response = new Response<bool, string>
+                    {
+                        Status = Response<bool, string>.Success(),
+                        Value = true
+                    };
+                }
+                else
+                {
+                    string msgResponse = string.Format("ContributionData:{0}", metaUploadResponse.msg);
+                    response = new Response<bool, string>
+                    {
+                        Status = Response<bool, string>.Failed(new Exception(msgResponse)),
+                        ErrorValue = msgResponse
+                    };
+                }
+                return response;
+            }
+            catch (Exception ex)
+            {
+                return new Response<bool, string>
+                {
+                    Status = Response<bool, string>.Failed(ex),
+                    ErrorValue = ex.Message
+                };
+            }
+        }
+
+        private static async Task<(bool status, string msg)> ContentsUpload(HttpClient client, string endPoint, string uploadName, MultipartFormDataContent contents)
```

Comment:
Is it possible to consolidate these into a function that takes a HttpContent?

Reply:
Hi Chris, yes, I added new commit about this (fix: consolidate contentsupload functions).